build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/extractor/GoZipBallStreamer.java

```diff
@@ -238,29 +244,59 @@ private boolean isSubFolderOfAnotherModule(String moduleRootDir, String director
      * @return True if the entry is a go.mod file which doesn't belong to the project we are packing
      */
     private boolean isSubModule(String entryName) {
-        if (entryName.endsWith(MOD_FILE)) {
+        if (entryName.endsWith(MOD_FILE_PATH)) {
             if (shouldPackSubModule()) {
-                return (!entryName.substring(entryName.indexOf('/') + 1).endsWith(subModuleName + MOD_FILE));
+                return (!entryName.substring(entryName.indexOf('/') + 1).endsWith(subModuleName + MOD_FILE_PATH));
             } else {
-                return entryName.substring(entryName.indexOf('/') + 1).endsWith(subModuleName + MOD_FILE);
+                return !entryName.substring(entryName.indexOf('/') + 1).equals(MOD_FILE);
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @param majorVersion - major version of a compatible v2 project
+     * @return true, if the go.mod file in the root specify a path of submodule of the given major version
```

Comment:
specify --> specifies

Since this description is not easy to understand, I suggest adding an example to the description.